File: internal/collect/ollama_test.go
package collect

import (
	"reflect"
	"testing"
)

func TestParseOllamaList(t *testing.T) {
	const real = "NAME              ID            SIZE      MODIFIED\n" +
		"llama3.2:latest   a80c4f17acd5  2.0 GB    2 weeks ago\n" +
		"qwen2.5:7b        845dbda0ea48  4.7 GB    3 months ago"

	tests := []struct {
		name string
		in   string
		want []OllamaModel
	}{
		{
			name: "parses rows, skips header, drops ID, keeps spaced values",
			in:   real,
			want: []OllamaModel{
				{Name: "llama3.2:latest", Size: "2.0 GB", Modified: "2 weeks ago"},
				{Name: "qwen2.5:7b", Size: "4.7 GB", Modified: "3 months ago"},
			},
		},
		{
			name: "header only",
			in:   "NAME  ID  SIZE  MODIFIED",
			want: nil,
		},
		{
			name: "empty",
			in:   "",
			want: nil,
		},
		{
			name: "whitespace only",
			in:   "   \n  ",
			want: nil,
		},
		{
			name: "row with leading/trailing whitespace and blank line",
			in:   "NAME  ID  SIZE  MODIFIED\n  mistral:7b  abc123  4.1 GB  5 days ago  \n\n",
			want: []OllamaModel{
				{Name: "mistral:7b", Size: "4.1 GB", Modified: "5 days ago"},
			},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := ParseOllamaList(tt.in)
			if !reflect.DeepEqual(got, tt.want) {
				t.Errorf("ParseOllamaList() = %#v, want %#v", got, tt.want)
			}
		})
	}
}

File: tests/collectors/ollama.test.ts
import { test, expect, describe } from "bun:test";
import { parseOllamaList, makeOllamaCollector } from "../../src/collectors/ollama";
import type { CollectorContext } from "../../src/collectors/types";
import { fakeEnv } from "../helpers/fake-env";

const ctx: CollectorContext = { redact: true, home: "/fake/home" };

// `ollama list` columns: NAME  ID  SIZE  MODIFIED (2+ spaces between columns).
const real = `NAME              ID            SIZE      MODIFIED
llama3.2:latest   a80c4f17acd5  2.0 GB    2 weeks ago
qwen2.5:7b        845dbda0ea48  4.7 GB    3 months ago`;

describe("parseOllamaList", () => {
  test("parses rows, skips header, drops the ID column, keeps spaced values", () => {
    expect(parseOllamaList(real)).toEqual([
      { name: "llama3.2:latest", size: "2.0 GB", modified: "2 weeks ago" },
      { name: "qwen2.5:7b", size: "4.7 GB", modified: "3 months ago" },
    ]);
  });

  test("header only → []", () => {
    expect(parseOllamaList("NAME  ID  SIZE  MODIFIED")).toEqual([]);
  });

  test("empty → []", () => {
    expect(parseOllamaList("")).toEqual([]);
  });
});

describe("makeOllamaCollector", () => {
  test("builds ai.ollama.models with name | size | modified rows", async () => {
    const r = await makeOllamaCollector(fakeEnv({ outputs: { ollama: real } }))(ctx);
    expect(r["ai.ollama.models"]?.items).toEqual([
      { raw: "llama3.2:latest | 2.0 GB | 2 weeks ago", columns: ["llama3.2:latest", "2.0 GB", "2 weeks ago"] },
      { raw: "qwen2.5:7b | 4.7 GB | 3 months ago", columns: ["qwen2.5:7b", "4.7 GB", "3 months ago"] },
    ]);
  });

  test("ollama not installed (throws) → {}", async () => {
    const r = await makeOllamaCollector(
      fakeEnv({
        run: () => {
          throw new Error("command not found: ollama");
        },
      }),
    )(ctx);
    expect(r).toEqual({});
  });

  test("no models → {}", async () => {
    expect(await makeOllamaCollector(fakeEnv({ outputs: { ollama: "" } }))(ctx)).toEqual({});
  });
});
